### DISCUSSION TITLE
Retrieve Activity Logs on a Model with LogActivity Trait

### DISCUSSION THREAD

**Discussion by FallenKnight85** [spatie/laravel-activitylog#750] [Q&A] [answered]
Hi, been trying to find in your docs on how do I retrieve the logs on a Specific Model? (it's always "get last activity" on examples but I'm looking for enumerating logs that have been performed on that specific model)

I'm not getting results from

return $model->activity;

Do I need to define the relationship of the Model with the Activity Model?

My aim is to get logs of the model and show the causer and changes made by the causer.

**Comment by danord24** [comment]
Hi, you can do: `Activity::where('subject_type', YourModel::class)->with('causer')->get();`

Hope that helps.

**Comment by Gummibeer** [comment, marked as answer]
At first, I wonder how you've created this as an issue? The only way I've found would be that you've used the "Bug" template and removed the label? Or have you accessed https://github.com/spatie/laravel-activitylog/issues/new directly? Or is there another way GitHub does not cover with the current settings?
At all, we have several issue/discussion templates you can choose one: https://github.com/spatie/laravel-activitylog/issues/new/choose
I kindly ask you to use the correct one in the future. For the moment I've moved the issue to the "Help Discussion".

------

From the subject model perspective you can use the `$subject->activities` relationship.
https://github.com/spatie/laravel-activitylog/blob/b07e562babd3d88d23350f8ce5453e49add4c33c/src/Traits/LogsActivity.php#L80-L83

If you want to show all activities done by an user you can use the `$user->actions` relationship if your user model uses the `CausesActivity` trait.
https://github.com/spatie/laravel-activitylog/blob/b07e562babd3d88d23350f8ce5453e49add4c33c/src/Traits/CausesActivity.php#L10-L16

**Comment by FallenKnight85** [comment]
Thank you! Sorry, I'm new to Github, I thought I should put my questions to issues. 

```php
public function showLogs(Request $request, CollegeEnrollment $collegeEnrollment)
    {
        if(Auth::user()->can('college.enrollment.admin')) {
            $collegeEnrollment->activities->load('causer');
            return $collegeEnrollment->activities;
        }
    }
```

Thank you so much for your help! It works now!

**Reply by Gummibeer** [reply]
Yes you can use issues, but if you click on "new issue" button you should see https://github.com/spatie/laravel-activitylog/issues/new/choose with a list of options (bug, question, idea ...) - if you choose the one that matches your case you will get to the corresponding form. Only bugs are in issues right now. This is to keep the real issues clean (like a jira board) and everything not really a task will be in discussions.
That's not about any different importance to us. Only to have a better overview of open tasks that require code changes and open discussions waiting for help, feedback.

And happy that it worked! 🎉
